libfabric.spec

```diff
@@ -85,69 +90,104 @@ Development files for the libfabric library.
 %endif
 
 %prep
-%autosetup -p1 -n libfabric-%dl_version
+%autosetup -p1 -n %{name}-%{dl_version}
 
 %build
+%if %{__remake_config}
+./autogen.sh
+%endif
 %if (0%{?suse_version} > 0)
 export CFLAGS="%{optflags} -fPIC -pie"
 export CXXFLAGS="%{optflags} -fPIC -pie"
 %endif
-if [ ! -f configure ]; then
-    ./autogen.sh
-fi
 # defaults: with-dlopen can be over-rode:
-%configure %{?_without_dlopen} %{configopts} \
-%ifnarch s390
-        --with-valgrind
+%configure  --disable-static          \
+            --disable-silent-rules    \
+            %{?_without_dlopen}       \
+            --with-valgrind           \
+            --enable-sockets          \
+            --enable-tcp              \
+            --enable-verbs            \
+            --enable-rxm              \
+            --enable-shm              \
+%if 0%{?rhel} >= 8 || 0%{?suse_version} >= 1315
+            --enable-opx              \
+%else
+            --disable-opx             \
 %endif
+            --disable-usnic           \
+            --disable-efa             \
+            --disable-dmabuf_peer_mem \
+            --disable-hook_hmem       \
+            --disable-hook_debug      \
+            --disable-trace           \
+            --disable-perf            \
+            --disable-rstream         \
+            --disable-rxd             \
+            --disable-mrail           \
+            --disable-udp             \
+            --disable-psm             \
+            --disable-psm2            \
+            --disable-psm3            \
+            --disable-gni             \
+            --disable-bgq
+
 sed -i 's|^hardcode_libdir_flag_spec=.*|hardcode_libdir_flag_spec=""|g' libtool
 sed -i 's|^runpath_var=LD_RUN_PATH|runpath_var=DIE_RPATH_DIE|g' libtool
+%make_build
 
-make %{?_smp_mflags} V=1
 
 %install
 %make_install
-# remove unpackaged files from the buildroot
-rm -f %{buildroot}%{_libdir}/*.la
-%fdupes %{buildroot}/%{_prefix}
```

Comment:
Removing this is generating a truck-load of duplicate file warnings from `rpmlint`.

Reply:
yes I do see that, I don't understand why this is needed though. There are no duplicates as far as I can tell and it seems like fdupes might be doing something we don't want it to do ? should libfabric upstream be patched instead ? do you have any idea of what libfabric is doing wrong in that case ?

Reply:
```
libfabric-devel.x86_64: W: files-duplicate /usr/share/man/man3/fi_shutdown.3.gz /usr/share/man/man3/fi_accept.3.gz:/usr/share/man/man3/fi_connect.3.gz:/usr/share/man/man3/fi_getname.3.gz:/usr/share/man/man3/fi_getpeer.3.gz:/usr/share/man/man3/fi_join.3.gz:/usr/share/man/man3/fi_leave.3.gz:/usr/share/man/man3/fi_listen.3.gz:/usr/share/man/man3/fi_reject.3.gz:/usr/share/man/man3/fi_setname.3.gz
```

Looks like perhaps they are all sharing the same manpage perhaps, yet doing it by making copies at `make install` time rather than creating hard links.

```
fi_cm(3)                           #VERSION#                          fi_cm(3)

NAME
       fi_cm - Connection management operations

       fi_connect / fi_listen / fi_accept / fi_reject / fi_shutdown
              Manage endpoint connection state.

       fi_setname / fi_getname / fi_getpeer
              Set local, or return local or peer endpoint address.

       fi_join / fi_close / fi_mc_addr
              Join, leave, or retrieve a multicast address.

SYNOPSIS
              #include <rdma/fi_cm.h>

              int fi_connect(struct fid_ep *ep, const void *addr,
                  const void *param, size_t paramlen);

              int fi_listen(struct fid_pep *pep);

              int fi_accept(struct fid_ep *ep, const void *param, size_t paramlen);

              int fi_reject(struct fid_pep *pep, fid_t handle,
                  const void *param, size_t paramlen);

              int fi_shutdown(struct fid_ep *ep, uint64_t flags);

              int fi_setname(fid_t fid, void *addr, size_t addrlen);

              int fi_getname(fid_t fid, void *addr, size_t *addrlen);

              int fi_getpeer(struct fid_ep *ep, void *addr, size_t *addrlen);

              int fi_join(struct fid_ep *ep, const void *addr, uint64_t flags,
                  struct fid_mc **mc, void *context);

              int fi_close(struct fid *mc);

              fi_addr_t fi_mc_addr(struct fid_mc *mc);
```

So yeah, technically, I suppose this could be addressed upstream by having their `make install` create hardlinks instead of copies, but `%fdupes` takes care of it if they don't want to do that.

Reply:
Interesting thanks, I'll have a look. If I see that it's just a nightmare to fix, I'll put back the fdupes.

Reply:
I reverted that change back. All the manpage stuff seems to be built around those dummy manpages that get installed so would be some rework to fix it properly.